zuul-core/src/main/java/com/netflix/zuul/netty/server/push/PushMessageSenderInitializer.java

```diff
@@ -26,20 +26,13 @@
  * Date: 5/16/18
  */
 public abstract class PushMessageSenderInitializer extends ChannelInitializer<Channel> {
-
-    private final PushConnectionRegistry pushConnectionRegistry;
-
-    public PushMessageSenderInitializer(PushConnectionRegistry pushConnectionRegistry) {
-        this.pushConnectionRegistry = pushConnectionRegistry;
-    }
-
     @Override
     protected void initChannel(Channel ch) throws Exception {
         final ChannelPipeline pipeline = ch.pipeline();
         pipeline.addLast(new HttpServerCodec());
         pipeline.addLast(new HttpObjectAggregator(65536));
-        pipeline.addLast(getPushMessageSender(pushConnectionRegistry));
+        addPushMessageHandlers(pipeline);
     }
 
-    protected abstract PushMessageSender getPushMessageSender(PushConnectionRegistry pushConnectionRegistry);
+    protected abstract void addPushMessageHandlers(final ChannelPipeline pipeline);
```

Comment:
Any concerns about this being a breaking change for folks? Not sure if you'd want to have a separate function to add other handlers instead of replacing, but I'll leave it up to you.